ast/src/object/mod.rs

```diff
@@ -24,4 +24,47 @@ pub struct Object<T> {
     pub degree: u64,
     /// the pil identities for this machine
     pub pil: Vec<PilStatement<T>>,
+    /// the links from this machine to its children
+    pub links: Vec<Link<T>>,
+}
+
+#[derive(Clone)]
+pub struct Link<T> {
+    pub from: LinkFrom,
+    pub to: LinkTo<T>,
+}
+
+#[derive(Clone)]
+pub struct LinkFrom {
+    pub instr: Instr,
+}
+
+#[derive(Clone)]
+pub struct LinkTo<T> {
+    /// the location of the machine we link to
+    pub loc: String,
+    /// its latch
+    pub latch: String,
+    /// the function we link to
+    pub function: Function<T>,
+}
+
+#[derive(Clone)]
+pub struct Instr {
+    pub name: String,
+    pub params: Params,
+}
+
+impl Instr {
+    /// the flag for this instruction
+    pub fn flag(&self) -> String {
+        format!("instr_{}", self.name)
```

Comment:
is this the column name it compiles to?

Reply:
Yes

Reply:
so this relies on the code gen also doing the same thing?

Reply:
I reverted this as it was confusing

Reply:
how is the same achieved now?

Reply:
Oh, with a struct field instead of a method.